### DISCUSSION TITLE
Use with XML?

### DISCUSSION THREAD

**Discussion by ryanwclark1** [scikit-hep/awkward#841] [Q&A]
I will preface this by saying I am exploring the library and I'm not a current user and have only skimmed the documentation.  Does awkward have a straight forward approach for handling XML?  I found the from json/parquet/arrow functions and although I didn't see an explicit from_xml didn't know if someone with more experience with the library could point me in the right direction.  Thank you in advance.

**Comment by jpivarski** [comment]
It currently doesn't, though XML is a natural way to express nestedness relationships. You haven't missed anything in the docs, though.

You might want to consider building Awkward Arrays from XML using [ak.ArrayBuilder](https://awkward-array.readthedocs.io/en/latest/_auto/ak.ArrayBuilder.html). That's how the JSON is implemented (though in C++, but Python should be fast enough if the source format is text).
